tests/flakes-with-submodules.sh

```diff
@@ -0,0 +1,149 @@
+source common.sh
+
+if [[ -z $(type -p git) ]]; then
+    echo "Git not installed; skipping flake tests"
+    exit 99
+fi
+
+clearStore
+rm -rf $TEST_HOME/.{cache,config}
+
+registry=$TEST_ROOT/registry.json
+
+nonFlakeDir=$TEST_ROOT/nonFlake
+flakeDir=$TEST_ROOT/flake
+flakeWithSubmodules=$TEST_ROOT/flakeWithSubmodules
+
+for repo in $flakeDir $nonFlakeDir $flakeWithSubmodules; do
+    rm -rf $repo $repo.tmp
+    mkdir -p $repo
+    git -C $repo init
+    git -C $repo config user.email "foobar@example.com"
+    git -C $repo config user.name "Foobar"
+    echo FNORD > $repo/README.md
+    git -C $repo add README.md
+    git -C $repo commit -m 'Initial'
+done
+
+cp config.nix $flakeDir
+
+cat > $flakeDir/flake.nix <<EOF
+{
+    description = "Flake whose defaultPackage prints out its README.md";
+
+    outputs = { self }: with import ./config.nix; let
+        inherit (self.modules) nonFlake;
+        writeShellScriptBin = name: script: mkDerivation {
+            inherit name;
+            buildCommand = ''
+                mkdir -p \$out/bin
+                bin=\$_/\${name}
+                echo '#!/bin/sh' > \$bin
+                echo '\${script}' >> \$bin
+                chmod +x \$bin
+            '';
+        };
+    in {
+        packages.$system = {
+            cat-own-readme = writeShellScriptBin "cat-own-readme" ''
+                cat \${self}/README.md
+            '';
+        };
+
+        defaultPackage.$system = self.packages.$system.cat-own-readme;
+    };
+}
+EOF
+
+git -C $flakeDir add .
+git -C $flakeDir commit -m'add flake.nix'
+
+[[ $(nix run $flakeDir#cat-own-readme) == "FNORD" ]]
+
+git -C $flakeWithSubmodules submodule add $nonFlakeDir
+git -C $flakeWithSubmodules submodule add $flakeDir
+git -C $flakeWithSubmodules add .
+git -C $flakeWithSubmodules commit -m'add submodules'
```

Comment:
If I were being a Mighty Stickler I'd request a test that declares this feature's behaviour in the presence of recursive submodules. (That is, repo A has a submodule B, which itself has a submodule C.)

Reply:
(Ah, I see you commented above that this isn't really supported. And anyway I wouldn't consider this blocking!)